_drafts/2021-12-13-promises-unfulfilled-handling-promise-errors.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: Handling Promise Errors
+author_avatar: https://avatars.githubusercontent.com/u/5316399?v=4
+author_name: Sam Messina
+author_github_username: regexpressyourself
+---
+
+In a recent Lunch and Learn with the Maxwell engineering team, we noticed two different ways to handle a rejected Promise. One can either include a second argument to the Promise's `.then()` function, or add a `.catch()` function. Curious to learn more, I decided to dig in further.
+
+![](/images/posts/2021-12-13-promises-unfulfilled-handling-promise-errors/detour.jpeg)
+
+## The two methods of catching rejected Promises
+
+**Method #1**: The first method to handle a rejected Promise involves adding a second callback to the `.then()` function:
+
+```javascript
+Promise.reject().then(
+  () => console.log('Success'),
+  () => console.log('Error')
+)
+```
+
+**Method #2**: The second method to handle a rejected Promise is adding a `.catch()` function:
+
+```javascript
+Promise.reject()
+  .then(() => console.log('Success'))
+  .catch(() => console.log('Error'))
+```
+
+
+## Similarities between the two methods
+
+The two methods of handling rejected Promises are almost entirely the same. From MDN:
+
+> ...calling obj.catch(onRejected) internally calls obj.then(undefined, onRejected).
+>
+> - [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch)
+
+In almost all cases, either method will work. There is one major exception, however.
+
+## Differences between the two methods
+
+Despite the nearly-identical implementations, there is one major difference. When either (1) an error is thrown in the `.then()` handler, or (2) the `.then()` handler returns a rejected Promise, only the `.catch()` function will  catch the error.
```

Comment:
I think you've got an extra space in between `will` and `catch`

Reply:
Ah, so I do. Nice "catch" 😄